packages/cardano-services/src/Projection/prepareTypeormProjection.ts

```diff
@@ -383,8 +385,10 @@ export const prepareTypeormProjection = (
     mappers: selectedMappers,
     stores: selectedStores,
     willStore: <T extends ProjectionEvent>(evt: T) =>
+      evt.eventType === ChainSyncEventType.RollBackward ||
       // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      evt.eventType === ChainSyncEventType.RollBackward || willStoreCheckers.some((check) => check(evt as any))
+      willStoreCheckers.some((check) => check(evt as any)) ||
+      willStoreBlockData(evt)
```

Comment:
Very minor improvement: I suggest making this the 1st condition (before `evt.eventType === ChainSyncEventType.RollBackward`), because this predicate is very cheap, and once it reaches tip-k it won't have to ever run any other predicates. If you make this change, feel free to squash right away.